OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -96,7 +96,11 @@ public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack =
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove, forceAttack);
+			var attack = attackBase.FirstOrDefault(x => !x.IsTraitDisabled && x.HasAnyValidWeapons(target));
+			if (attack == null)
+				throw new LuaException("Actor {0} was ordered to attack {1} but has no active Attack* trait to do so!.".F(Self, targetActor));
```

Comment:
I haven't looked at this pr properly, but at least this case seems to be written to only care about the "switch from one Attack* type to another" case.  We want to be able to support multiple active Attack*'s, so this should enumerate over all valid ones instead of picking the first, and not throw an exception if there are none (you could return a bool instead).